src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -22,11 +22,14 @@ internal partial class EndpointHtmlRenderer
     private HashSet<int>? _visitedComponentIdsInCurrentStreamingBatch;
     private string? _ssrFramingCommentMarkup;
     private bool _isHandlingErrors;
+    private bool _hasStatusCodePage;
 
-    public void InitializeStreamingRenderingFraming(HttpContext httpContext, bool isErrorHandler)
+    public void InitializeStreamingRenderingFraming(HttpContext httpContext, bool isErrorHandler, bool hasStatusCodePage)
     {
         _isHandlingErrors = isErrorHandler;
-        if (IsProgressivelyEnhancedNavigation(httpContext.Request))
+        _hasStatusCodePage = hasStatusCodePage;
+        bool avoidEditingHeaders = hasStatusCodePage && httpContext.Response.StatusCode == StatusCodes.Status404NotFound;
```

Comment:
Is this effectively disabling streaming renders in 404 responses? Is there a reason for us to do it?

Reply:
Why it's needed: without it, we will add the dictionary key twice.

Case when we enter that condition:
- the page that sets 404 is requested with enhance nav, e.g. we used `NavigationManager.NavigateTo("page-that-sets-not-found")`.

The re-executed page is inheriting the headers of the original page, so `IsProgressivelyEnhancedNavigation` will return true. If we allow entering the condition, we will get an exception because we already edited the headers in the original page. 

```
ArgumentException: An item with the same key has already been added. Key: ssr-framing
```